apps/software/lib/software/model/storage.ex

```diff
@@ -28,19 +29,6 @@ defmodule Helix.Software.Model.Storage do
   end
 
   @spec create_changeset() :: Ecto.Changeset.t
-  def create_changeset do
-    %__MODULE__{}
-    |> cast(%{}, [])
-    |> put_primary_key()
-  end
-
-  @spec put_primary_key(Ecto.Changeset.t) :: Ecto.Changeset.t
-  defp put_primary_key(changeset) do
-    if get_field(changeset, :storage_id) do
-      changeset
-    else
-      pk = PK.pk_for(__MODULE__)
-      put_change(changeset, :storage_id, pk)
-    end
-  end
+  def create_changeset,
+    do: cast(%__MODULE__{}, %{}, [])
```

Comment:
>Also, how dialyzer behaves with default fields? It does also require a typespec like
> ```
> @spec create_changeset() :: Ecto.Changeset.t
> ```
> To work properly?

When you write
```elixir
foo(params \\ []), do: bar(params)
```
it is compiled to
```erlang
foo() ->
  foo([]).
foo(X) ->
  bar(x).
```

You actually don't need to write any typespec because dialyzer will check `foo() will call foo with [] as args, what will foo([]) return ? that is certainly the return of foo()`

So, you *can* write specs for the function without the defaults but if you write the spec for the full-fledged function you are automatically speccing it in all cases